### ISSUE TITLE
PDF export dialog

### ISSUE THREAD

**Issue by letsfindaway** [letsfindaway/OpenBoard#100]
To give the user a better control over PDF export I propose to add a PDF export dialog in the workflow. This dialog bundles various adjustments of paper size, scaling and background export. As a first sketch this dialog may look as follows:
![grafik](https://user-images.githubusercontent.com/35591413/193069740-c03ce10a-872b-447e-becf-10cacfc8b85a.png)
I n the left area you can configure a paper size and margins. The bottom left contains the options to export the background color or grid. The top right gives some options on how the paper size should be handled:

- Shall we always try to fill the paper or shall we try to observe the given scale?
- If a page contains a PDF background base document, shall we use that for the paper size?
- If the content would not fit, shall we scale it down or extend the paper size?

The bottom right gives a page preview where you can flip through the pages to see the result.

The file selection is prefilled to the previously used path and the document name. To change this, press the "Choose" button to the right.

**Comment by letsfindaway** [comment]
More ideas:

- Probably better move the output file selection pane to the bottom of the dialog.
- Add a third option `Auto` to the Portrait and Landscape selection, which chooses the orientation based on the size ratio of the content.

**Comment by kaamui** [comment]
I see that you're doing a lot of work to improve OpenBoard, and this is great, but keep in mind that the Openboard tends to remain simple. I would not you to build something too huge, put a lot of effort in it, for it not to be implemented in the end because it didn't convinced the whole decision board. 

If you aware of this, but are planning this kind of improvement only for openSUSE, or for community driven packages, it's great. Just wanted to be sure that you're OK with this.

**Comment by letsfindaway** [comment]
This is currently just some exercise. Beside the form, I have not created anything. But it is also a good way to visualize the options you have - even if you would not create that dialog. But I still feel that you have different requirements for the exported PDF, even within you customers:

- Do they want to **create some printable handout**? Then it would be best to have everything in DIN A4, Landscape or Portrait as it fits better. And some margins would also be good, so that the result can be punched and filed.
- Do they want to **annotate an imported PDF**? Then it would be best if the output page sizes match the input page sizes.
- Or should the result be used for a **PDF presentation**? Then everything has to have Landscape and a fixed size of probably 16:9.

If you don't let them choose, you can never fulfill all wishes. So I still think this would be useful.

And no: if it has no chance to ever go into `master`, then I stop working on this. I don't want to make a fork of OpenBoard with too much features added. But still keep my thoughts here. Probably they may get interesting later ;)

**Comment by kaamui** [comment]
Yep. To be clear, I'm not against the idea personally, I just wanted to remind you that I'm not alone taking decisions, and that this kind of features actually doesn't match with the ideology of "keep it simple" that made OpenBoard exist.

Historically, OpenBoard was created to take the opposite direction of Open-Sankoré, that was getting more complex release after release. And we have multiple archetypes of users, of hardware associated with OpenBoard, and even multiple class context, so you'll never fulfill all wishes :)

**Comment by letsfindaway** [comment]
> Yep. To be clear, I'm not against the idea personally, I just wanted to remind you that I'm not alone taking decisions, and that this kind of features actually doesn't match with the ideology of "keep it simple" that made OpenBoard exist.
> 
> Historically, OpenBoard was created to take the opposite direction of Open-Sankoré, that was getting more complex release after release. And we have multiple archetypes of users, of hardware associated with OpenBoard, and even multiple class contexts, so you'll never fulfill all wishes :)

I completely understand this. And I still keep thinking about some kind of compromise. One idea would be to have the three scenarios I described above as some simple choice for the user, e.g. directly in the Export to PDF menu:

- Export as handout
- Export as presentation
- Export as annotated PDF

and to have reasonable settings behind these options, which can eventually be tuned using configuration variables.
